GDJS/GDJS/IDE/ExporterHelper.cpp

```diff
@@ -277,6 +293,36 @@ gd::String ExporterHelper::ExportProjectData(
   return "";
 }
 
+void ExporterHelper::SerializeUsedResources(
+    gd::SerializerElement &rootElement,
+    std::set<gd::String> &projectUsedResources,
+    std::vector<std::set<gd::String>> &scenesUsedResources) {
+
+  auto serializeUsedResources =
+      [](gd::SerializerElement &element,
+         std::set<gd::String> &usedResources) -> void {
+    auto &resourcesElement = element.AddChild("usedResources");
+    resourcesElement.ConsiderAsArrayOf("resourceReference");
+    for (auto &resourceName : usedResources) {
+      auto &resourceElement = resourcesElement.AddChild("resourceReference");
+      resourceElement.SetAttribute("name", resourceName);
+    }
+  };
+
+  serializeUsedResources(rootElement, projectUsedResources);
+
+  auto &layoutsElement = rootElement.GetChild("layouts");
+  std::size_t layoutCount =
+      std::min(scenesUsedResources.size(), layoutsElement.GetChildrenCount());
+  for (std::size_t layoutIndex = 0; layoutIndex < layoutCount; layoutIndex++) {
+    // Both lists should have the same order.
```

Comment:
Ideally you would avoid this by using a unordered_map, which maps from scene name to the used resources.